### ISSUE TITLE
Why is there a difference in allocated memory between malloc_stats_print and jeprof top?

### ISSUE THREAD

**Issue by jjoo172** [jemalloc/jemalloc#1507]
Hello,

I am using heap dumps to analyze memory usage with jemalloc, and I'm seeing a very large discrepancy in allocated memory between what malloc_stats_print() reports vs. what I get when I call top when using jeprof.

For example, in malloc_stats_print I see something like:

```___ Begin jemalloc statistics ___
Version: 5.0.1-0-g896ed3a8b3f41998d4fb4d625d30ac63ef2d51fb
Assertions disabled
config.malloc_conf: ""
Run-time option settings:
  opt.abort: false
  opt.abort_conf: false
  opt.retain: true
  opt.dss: "secondary"
  opt.narenas: 160
  opt.percpu_arena: "disabled"
  opt.background_thread: true (background_thread: true)
  opt.dirty_decay_ms: 10000 (arenas.dirty_decay_ms: 10000)
  opt.muzzy_decay_ms: 10000 (arenas.muzzy_decay_ms: 10000)
  opt.junk: "false"
  opt.zero: false
  opt.tcache: true
  opt.lg_tcache_max: 15
  opt.prof: true
  opt.prof_prefix: "jeprof"
  opt.prof_active: false (prof.active: true)
  opt.prof_thread_active_init: true (prof.thread_active_init: true)
  opt.lg_prof_sample: 19 (prof.lg_sample: 19)
  opt.prof_accum: false
  opt.lg_prof_interval: -1
  opt.prof_gdump: false
  opt.prof_final: false
  opt.prof_leak: false
  opt.stats_print: false
Arenas: 160
Quantum size: 16
Page size: 4096
Maximum thread-cached size class: 32768
Allocated: 11075266920, active: 12517670912, metadata: 1221200320, resident: 13926576128, mapped: 14209425408, retained: 88352841728
```
whereas when I run jeprof using a heap dump from the same time period I get something like:
`(jeprof) top
Total: 5433.0 MB`

Is there a reason why this discrepancy might exist? Is fragmentation accounted for in one but not the other? Thank you in advance!

**Comment by interwq** [comment]
Answered in gitter: 
The number showed in jeprof is based on the sampled allocations. When enabled, approxiamately one out of every 2<<lg_prof_sample bytes of allocaitons will be sampled and reported in the heap profile. See http://jemalloc.net/jemalloc.3.html#opt.lg_prof_sample

**Comment by jjoo172** [comment]
So I actually did a bit of an investigation and found that the memory gap seemed to be due to heap profiling itself, since reducing sampling interval increased the gap. In other words, in my experiment, I saw that when we sampled very frequently, we would see that the heap stats used memory numbers went very high, and that the actual number of bytes in the heap dump decreased significantly. That leads to the following question - is there a way to tell (either from heap stats or from some other source) how much memory the profiling itself is consuming?
